src/core/datatypes/trees/Tree.cpp

```diff
@@ -2052,10 +2051,24 @@ TopologyNode& Tree::reverseParentChild(TopologyNode &n)
         TopologyNode &p = n.getParent();
         ret = &(reverseParentChild(p));
 
+        // save p values
+        size_t idx_p = p.getIndex();
+        double brlen_p = p.getBranchLength();
+        std::vector<std::string> ncomm_p = p.node_comments;
+        std::vector<std::string> bcomm_p = p.branch_comments;
+
         // we need to re-orient the branches/indices so that
         // nodes remain associated with the same parameters
```

Comment:
Maybe we should change this to say that we reorient so that _branches_ remain associated with the same parameters: index, branch_length, and comments.

Reply:
Long-term we should probably add a separate index for the node that does not get swapped, so that we can have consistent node indices.  Then we can call the current index the branch index.

Reply:
I agree long-term that's a better solution re: branch vs. node indices. And yeah, I can change that comment, it's very confusing anyway